### PR TITLE
Squid API compute additions

### PR DESCRIPTION
**Pull request by kremalicious**
Additions to make the compute feature more useful. Essentially:

- ocean.compute.order
- ocean.compute.start
- ocean.compute.stop
- ocean.compute.restart
- ocean.compute.status
- ocean.compute.result
- ocean.compute.delete

Also prepare to include this spec in the docs under `/architecture`

### CODE REVIEW

File: doc/architecture/squid.md [hunk 1]
```diff
@@ -559,25 +476,213 @@ const success = ocean.accounts.requestTokens(account, amount)
...
+Returns information about the status of a compute job
+
+_Parameters_
+
+```
+agreementId: str -- The ID of the service agreement.
+jobId: str -- The ID of the job.
+```
+
+_Returns_
```
Comment (kremalicious) [inline doc/architecture/squid.md]: the status responses should be defined further

Comment (ssallam) [reply doc/architecture/squid.md]: Agreed.

Comment (kremalicious) [reply doc/architecture/squid.md]: defined based on Brizo API status responses https://github.com/oceanprotocol/brizo/blob/feature/compute/API.md

**Comment by alexcos20** [comment]
what shoud be the default values for publishAlgorithmLog and publishOutput?
It makes sense to have them both true and fill up the required URL with current this.config. states ?

**Review by ssallam — APPROVED** [review]

**Pull request merged**